pkg/trace/api/dogstatsd.go

```diff
@@ -0,0 +1,41 @@
+package api
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/log"
+)
+
+// dogstatsdProxyHandler returns a new HTTP handler which will proxy requests to the DogStatsD
+// endpoint in the Core Agent over UDP.
+func (r *HTTPReceiver) dogstatsdProxyHandler() http.Handler {
+	if !r.conf.StatsdEnabled {
+		log.Info("DogstatsD disabled in the Agent configuration")
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			http.Error(w, "405 Method Not Allowed", http.StatusMethodNotAllowed)
+		})
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		// TODO: Simplify this. We shouldn't need to open a new connection for every
```

Comment:
Maybe this is a TODO for later in the same PR, but why not do this here? Shouldn't be such a big deal.

Reply:
You reviewed a bit prematurely :) I'm still working on the changes. This was just posted up so the NodeJS team could try it while testing their own code.

Reply:
It doesn't look like this will be as straightforward as I had hoped. What I hoped I could do was manually set the `Transport` on a `http.ReverseProxy` to dial directly, but that fails as `unix` isn't a supported scheme. It shouldn't be too much of a performance issue to open a new connection each time the endpoint is hit, but we'll see this better in the end-to-end tests once we have a Node client working.

Reply:
Sorry. Nevermind me then 🙂 

![homer-simpson-homer](https://user-images.githubusercontent.com/6686356/189666135-6959ec5a-f239-4e23-b342-35b3b1f7823b.gif)

Reply:
No worries 😄